### DISCUSSION TITLE
Undefined symbols on Xcode

### DISCUSSION THREAD

**Discussion by konguele** [apache/cordova#455] [Q&A]
Good Morning,

I created my game in Gdevelop and when I use Cordova and CocoaPods, I get the following error on Xcode. It has the necessary frameworks (at least it has downloaded them correctly) and I see that they are arm64. Could you help me?

Gdevelop version: 5.2.176
Cordova Version: 12.0.0 (Cordova-lib@12.0.1)
Cocoapods version: 1.14.3

Thank you very much!

ld: Undefined symbols:
  _GULIsLoggableLevel, referenced from:
      -[APMMonitor isLoggableLevel:] in GoogleAppMeasurement[arm64][66](APMMonitor.o)
  _GULLogBasic, referenced from:
      -[APMASLLogger logMessage:logTag:messageCode:withLogLevel:] in GoogleAppMeasurement[arm64][11](APMASLLogger.o)
  _GULLogError, referenced from:
      -[APMPersistentDictionary initWithFileName:] in GoogleAppMeasurement[arm64][223](APMPersistentDictionary.o)
      _APMWriteDictionaryToURL in GoogleAppMeasurement[arm64][223](APMPersistentDictionary.o)
      -[APMUserDefaults synchronize] in GoogleAppMeasurement[arm64][274](APMUserDefaults.o)
  _GULLogInfo, referenced from:
      ___44+[UIViewController(APMScreenClassName) load]_block_invoke in GoogleAppMeasurement[arm64][291](UIViewController+APMScreenClassName.o)
  _GULLogWarning, referenced from:
      +[APMMeasurement sharedInstance] in GoogleAppMeasurement[arm64][61](APMMeasurement.o)
      -[APMPersistentDictionary objectForKey:] in GoogleAppMeasurement[arm64][223](APMPersistentDictionary.o)
      -[APMPersistentDictionary setObject:forKey:] in GoogleAppMeasurement[arm64][223](APMPersistentDictionary.o)
      -[APMUserDefaults objectForKey:] in GoogleAppMeasurement[arm64][274](APMUserDefaults.o)
      -[APMUserDefaults setObject:forKey:] in GoogleAppMeasurement[arm64][274](APMUserDefaults.o)
      -[APMUserDefaults synchronize] in GoogleAppMeasurement[arm64][274](APMUserDefaults.o)
      ___44+[UIViewController(APMScreenClassName) load]_block_invoke in GoogleAppMeasurement[arm64][291](UIViewController+APMScreenClassName.o)
      ...

**Comment by breautek** [comment]
Error states that you have undefined symbols.

Usually when you get to the linking step in compilation but fails because of undefined symbols, it means you have the headers (e.g. the compilation translation unit can compile the code based on the header information) but you're missing the library that actually implements those symbols/functions/classes. It could be that you're simply not linking against that library.

Based on the information provided, that's about all we can tell you. So I'd advise ensuring that you're actually linking against the library.

Inside xcode, when viewing your targets Build Phases, you can see what libraries are being linked with your library.

Also if you're developing on a M-chip mac and attempting to build/run in an iOS simulator. Try building and running on a physical iOS device instead. Not all google libraries actually supports or provide binaries for arm64 iOS simulators.
